backend-api/src/functions/services/session/tests/sessionService.test.ts

```diff
@@ -0,0 +1,304 @@
+  let sessionService: ISessionService;
+  let dynamoDbMockClient: AwsStub<
+    ServiceInputTypes,
+    ServiceOutputTypes,
+    DynamoDBClientResolvedConfig
+  >;
+
+  beforeEach(() => {
+    sessionService = new SessionService("mockTableName");
+    dynamoDbMockClient = mockClient(DynamoDBClient);
+  });
+
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date("2024-03-10"));
+  });
+
+  describe("Get active session ID", () => {
+    describe("Given an error happens when fetching an active session from the table", () => {
+      it("Returns an error response", async () => {
+        dynamoDbMockClient.on(QueryCommand).rejectsOnce("Mock DB Error");
+
+        const result = await sessionService.getActiveSessionId("mockSub");
+
+        expect(result.value).toStrictEqual({
+          errorMessage: "Error getting session ID - Error: Mock DB Error",
+          errorCategory: "SERVER_ERROR",
+        });
+        expect(result.isError).toBe(true);
+      });
+    });
+
+    describe("Given the query response does not contain the Items field", () => {
+      it("Returns success response with value of null", async () => {
+        dynamoDbMockClient.on(QueryCommand).resolvesOnce({});
+
+        const result = await sessionService.getActiveSessionId("mockSub");
+
+        expect(result.isError).toBe(false);
+        expect(result.value).toEqual(null);
+      });
+    });
+
+    describe("Given the query response does not return any items", () => {
+      it("Returns success response with value of null", async () => {
+        dynamoDbMockClient.on(QueryCommand).resolvesOnce({ Items: [] });
+
+        const result = await sessionService.getActiveSessionId("mockSub");
+
+        expect(result.isError).toBe(false);
+        expect(result.value).toEqual(null);
+      });
+    });
+
+    describe("Given the item returned is missing the attribute sessionId", () => {
+      it("Returns an error response", async () => {
+        dynamoDbMockClient
+          .on(QueryCommand)
+          .resolvesOnce({ Items: [{ dummyKey: { S: "dummyValue" } }] });
+
+        const result = await sessionService.getActiveSessionId("mockSub");
+
+        expect(result.isError).toBe(true);
+        expect(result.value).toEqual({
+          errorCategory: "SERVER_ERROR",
+          errorMessage: "Session is malformed",
+        });
+      });
+    });
+
+    describe("Given the session ID of an active session is found", () => {
+      it("Returns a success response with the session ID as value", async () => {
+        dynamoDbMockClient.on(QueryCommand).resolves({
+          Items: [
+            {
+              sessionId: { S: "mockSessionId" },
+            },
+          ],
+        });
+
+        const result = await sessionService.getActiveSessionId("mockSub");
+
+        expect(result.isError).toBe(false);
+        expect(result.value).toEqual("mockSessionId");
+      });
+    });
+  });
+
+  describe("Get active session details", () => {
```

Comment:
same on explicitly testing when two items are returned.

I'm starting to think this should be an error - as it implies we've let two users into the system when POST /credential should not create a new record if one already exists?